ros/api/v0/hosts.py

```diff
@@ -139,8 +139,15 @@ def build_sort_expression(self, order_how, order_method):
                 sort_order(PerformanceProfile.performance_score[
                     order_method].astext.cast(Integer)),
                 asc(PerformanceProfile.system_id),)
-        # FIXME: no ordering as of now for columns:
-        # state, recommendation_count
+
+        if order_method == 'recommendation_count':
+            return (sort_order(System.rule_hit_details),
+                    asc(PerformanceProfile.system_id),)
+
+        if order_method == 'state':
+            return (sort_order(System.rule_hit_details[0]['key']),
```

Comment:
Sure, I'll handle if no `rule_hit_details` are present. 
Well I'm not sure but, as per [#48](https://github.com/RedHatInsights/ros-backend/pull/48#discussion_r594150821), the rule_hit_details change inside systems table was merged there.

Reply:
@kgaikwad I did not get your point. What difference has a separate table for rule_hit_details have made? 
If having a separate `state` column in the system table is the suggestion(which will make sorting easy) then I am fine with that implementation.

Reply:
> If having a separate `state` column in the system table is the suggestion(which will make sorting easy) then I am fine with that implementation.

@patilsuraj767 - Yes, it will become easy. This was just a thought. 

As here sorting for `state` column is based on nested key of rule_hit_details, when no rule_hit_details captured for any system(s), will it simply return all records or give any warning/exception?
^^ this is what I wanted to confirm in my 1st query.
If it is going to return all records then @apuntamb you can skip this comment.

Reply:
Created `state` and `number_of_recommendations` columns in Systems table for easy sorting and searching (for later time).